types/mempool.go

```diff
@@ -0,0 +1,31 @@
+package types
+
+// MempoolTx we define an app-side mempool transaction interface that is as
+// minimal as possible, only requiring applications to define the size of the
+// transaction to be used when reaping and getting the transaction itself.
+// Interface type casting can be used in the actual app-side mempool implementation.
+type MempoolTx interface {
```

Comment:
Wish I had a chance to review this prior to merge. I really think this should not exist here.

IMO all mempool logic should live in a `mempool` subpackage in `baseapp/mempool`.